### PR TITLE
EDUCATOR-4408 bulk grades docs

### PR DESCRIPTION
**Pull request by Andytr1**
**Description:** Describe in a couple of sentences what this PR adds

**JIRA:** Link to JIRA ticket

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -13,11 +13,59 @@ Overview (please modify)
...
 
-The full documentation is at https://edx-bulk-grades.readthedocs.org.
+#. Clone this repo into ../src/ directory (relative to your "devstack" repo location). This will mount the directory in
+a way that is accessible to the lms container.
+
+#. From inside the lms, uninstall bulk-grades and reinstall your local copy. You can just copy the following line::
+
+    pip uninstall edx-bulk-grades -y; pip install -e /edx/src/edx-bulk-grades
+
+#. Now, get your completion development environment set up::
```
Comment (iloveagent57) [inline README.rst]: s/completion/bulk grades/

File: README.rst [hunk 2]
```diff
@@ -13,11 +13,59 @@ Overview (please modify)
...
+    source edx-bulk-grades-env/bin/activate
+    make requirements
+
+Making Code Changes
+-------------------
+1. After checking out a new branch, increment __version__ by the smallest possible value located in
+bulk_grades/__init__.py This will allow edx-platform to pick up the new version.
+2. Once a branch has been merged, it is necessary to make a release on github, specifying the new version from __version__
+set above.
+3. In order to platform to use the newest version of bulk-grades, it is necessary to run the::
```
Comment (iloveagent57) [inline README.rst]: "In order *for* platform..."

File: README.rst [hunk 3]
```diff
@@ -13,11 +13,59 @@ Overview (please modify)
...
+
+    $ make upgrade
+
+from docker shel of edx-platform. This will increment the version of edx-bulk-grades to the correct one.
```
Comment (iloveagent57) [inline README.rst]: s/shel/shell/

File: README.rst [hunk 4]
```diff
@@ -13,11 +13,59 @@ Overview (please modify)
...
+
+4. Once the code from step 3 is merged, this will trigger deployment of the correct versions of edx platform and bulk-grades.
+
+Unit Testing
+-------------------
+mock_apps folder: Since bulk_grades depends platform during actual runtime,
```
Comment (iloveagent57) [inline README.rst]: "depends *on*..."

File: README.rst [hunk 5]
```diff
@@ -13,11 +13,59 @@ Overview (please modify)
...
+ for unit tests, we need to mock various endpoints and calls. To this end, they are
+ mocked in in the mock_apps folder.
```
Comment (iloveagent57) [inline README.rst]: one too many "in"s

File: README.rst [hunk 6]
```diff
@@ -13,11 +13,59 @@ Overview (please modify)
 The ``README.rst`` file should then provide an overview of the code in this
 repository, including the main components and useful entry points for starting
 to understand the code in more detail.
+edx-bulk-grades is a library that runs under lms. It uses the configuration settings defined in lms as well.
+In order to use, the library must be installed
```
Comment (iloveagent57) [inline README.rst]: ...must be installed *into edx-platform*

File: README.rst [hunk 7]
```diff
@@ -7,17 +7,65 @@ edx-bulk-grades
...
-Documentation
--------------
+Using with Docker Devstack
+--------------------------
+Prerequisite: Have your Open edX https://github.com/edx/devstack properly installed.
+Note: When you see "from inside the lms" below, it means that you've run ``make lms-shell`` from your devstack directory
+and are on a command prompt inside the LMS container.
 
-The full documentation is at https://edx-bulk-grades.readthedocs.org.
+1. Clone this repo into ../src/ directory (relative to your "devstack" repo location). This will mount the directory in
```
Comment (iloveagent57) [inline README.rst]: Nit: you don't want a newline between "in" and "a way that..."

File: README.rst [hunk 8]
```diff
@@ -7,17 +7,65 @@ edx-bulk-grades
...
+3. Now, get your bulk-grades development environment set up::
+
+    cd /edx/src/edx-bulk-grades
+    virtualenv edx-bulk-grades-env
+    source edx-bulk-grades-env/bin/activate
+    make requirements
+
+Making Code Changes
+-------------------
+1. After checking out a new branch, increment __version__ by the smallest possible value located in
```
Comment (iloveagent57) [inline README.rst]: Nit: it's helpful to wrap "\_\_version\_\_" in double back-ticks, i.e. \`\`\_\_version\_\_\`\`

File: README.rst [hunk 9]
```diff
@@ -7,17 +7,65 @@ edx-bulk-grades
...
+bulk_grades/__init__.py This will allow edx-platform to pick up the new version.
```
Comment (iloveagent57) [inline README.rst]: Nit, same thing with back-ticks on `bulk_grades/__init__.py`

File: README.rst [hunk 10]
```diff
@@ -7,17 +7,65 @@ edx-bulk-grades
...
+2. Once a branch has been merged, it is necessary to make a release on github, specifying the new version from __version__
```
Comment (iloveagent57) [inline README.rst]: Nit: back-tick version again.

Comment (iloveagent57) [inline README.rst]: Nit: you need a newline before each item of a numbered list.

File: README.rst [hunk 11]
```diff
@@ -7,17 +7,65 @@ edx-bulk-grades
...
+set above.
+3. In order for platform to use the newest version of bulk-grades, it is necessary to run the::
```
Comment (iloveagent57) [inline README.rst]: Nit: newline before 3.

File: README.rst (hunk 7 above)
Comment (Andytr1) [reply README.rst]: Without a new line, it fails due to line being too long. How should I handle it?

Comment (iloveagent57) [reply README.rst]: Is it a quality check that fails?  If so, we should configure whatever is checking RST files in this repo to not worry about line length.

Comment (Andytr1) [reply README.rst]: Going to check it in without newline. Standby.

Comment (Andytr1) [reply README.rst]: Quality failure: 
docs run-test: commands[0] | doc8 --ignore-path docs/_build README.rst docs
Scanning...
Validating...
README.rst:25: D001 Line too long
README.rst:41: D001 Line too long
README.rst:43: D001 Line too long

Comment (iloveagent57) [reply README.rst]: Ah, I guess you want a "line block": http://docutils.sourceforge.net/docs/user/rst/quickref.html#line-blocks

something like this:
```
I'm going to number some things:

1. This is a normal length line.

2. | This is a super long line, it goes on and on and
   | onnn and on and on (notice the space before "onnn").

3. And here's a normal line again.

```

Try it out here: http://rst.aaroniles.net/

**Pull request merged**